### PR TITLE
Feature/generate_password

### PR DESCRIPTION
**Pull request by 24eeeee**
Added GeneratePasswordCommand unit test
Added endpoint GeneratePassword 
Added GeneratePassword command & interface

### CODE REVIEW

File: test/UserService.Business.UnitTests/GeneratePasswordCommandTests.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        private IGeneratePasswordCommand command;
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            command = new GeneratePasswordCommand();
+        }
+
+        [Test]
+        public void TEst()
```
Comment (TIove) [inline test/UserService.Business.UnitTests/GeneratePasswordCommandTests.cs:24]: Name it like other tests

**Review by TIove — CHANGES_REQUESTED** [review]

File: src/UserService.Business/Interfaces/IGeneratePasswordCommand.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace LT.DigitalOffice.UserService.Business.Interfaces
+{
+    public interface IGeneratePasswordCommand
+    {
```
Comment (djino97) [inline src/UserService.Business/Interfaces/IGeneratePasswordCommand.cs:4]: add xml comments

File: src/UserService/Controllers/UserController.cs [hunk 3]
```diff
@@ -36,6 +36,12 @@ public bool EditUser([FromServices] IEditUserCommand command, [FromBody] UserReq
             return command.Execute(request);
         }
 
+        [HttpGet("generatePassword")]
+        public void GeneratePassword([FromServices] IGeneratePasswordCommand command)
```
Comment (djino97) [inline src/UserService/Controllers/UserController.cs:40]: must be return password string

File: test/UserService.Business.UnitTests/GeneratePasswordCommandTests.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            for (int i = 0; i < 250; i++)
+            {
+                Assert.IsTrue(Regex.IsMatch(command.Execute(), "^(?=.*\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@!$_*#]).{8,12}$"));
```
Comment (djino97) [inline test/UserService.Business.UnitTests/GeneratePasswordCommandTests.cs:28]: ` "^(?=.*\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@!$_*#]).{8,12}$"`  - this needs to be separated into a variable

File: src/UserService.Business/GeneratePasswordCommand.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            needed[3] = SpecialSymbols[random.Next(SpecialSymbols.Length)];
+
+            var fillerLength = random.Next(4, 9);
+            var filler = new String(Enumerable.Repeat(TotalAlphabet, fillerLength)
+                .Select(s => s[random.Next(s.Length)])
+                .ToArray());
+
+            var result = new char[fillerLength + 4];
+            // randomly merging needed & filler into result
+            for (int i = 0, neededIndex = 0, fillerIndex = 0; i < result.Length; i++)
```
Comment (djino97) [inline src/UserService.Business/GeneratePasswordCommand.cs:32]: add empty lines

**Review by djino97 — CHANGES_REQUESTED** [review]

File: test/UserService.Business.UnitTests/GeneratePasswordCommandTests.cs (hunk 4 above)
Comment (DarkFelas) [reply test/UserService.Business.UnitTests/GeneratePasswordCommandTests.cs:28]: nope

File: src/UserService.Business/GeneratePasswordCommand.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+                    || neededIndex == needed.Length)
+                {
+                    result[i] = filler[fillerIndex];
+                    fillerIndex++;
+                }
+                else
+                {
+                    result[i] = needed[neededIndex];
+                    neededIndex++;
+                }
```
[the remark this answers is not in the document]
Comment (24eeeee) [reply src/UserService.Business/GeneratePasswordCommand.cs:44]: Should I add that class in the same file?

[the remark this answers is not in the document]
Comment (djino97) [reply src/UserService.Business/GeneratePasswordCommand.cs]: no, need separate

[the remark this answers is not in the document]
Comment (24eeeee) [reply src/UserService.Business/GeneratePasswordCommand.cs]: done

File: src/UserService.Business/GeneratePasswordCommand.cs (hunk 5 above)
Comment (24eeeee) [reply src/UserService.Business/GeneratePasswordCommand.cs]: done

File: src/UserService/Controllers/UserController.cs (hunk 3 above)
Comment (24eeeee) [reply src/UserService/Controllers/UserController.cs]: fixed

File: src/UserService.Business/Interfaces/IGeneratePasswordCommand.cs (hunk 2 above)
Comment (24eeeee) [reply src/UserService.Business/Interfaces/IGeneratePasswordCommand.cs:8]: done

File: test/UserService.Business.UnitTests/GeneratePasswordCommandTests.cs (hunk 1 above)
Comment (24eeeee) [reply test/UserService.Business.UnitTests/GeneratePasswordCommandTests.cs]: fixed

File: test/UserService.Business.UnitTests/GeneratePasswordCommandTests.cs (hunk 4 above)
Comment (24eeeee) [reply test/UserService.Business.UnitTests/GeneratePasswordCommandTests.cs:28]: idk, should i resolve it?

**Review by TIove — APPROVED** [review]

File: src/UserService.Business/GeneratePasswordCommand.cs [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            needed[1] = UpperCaseLetters[random.Next(UpperCaseLetters.Length)];
+            needed[2] = LowerCaseLetters[random.Next(LowerCaseLetters.Length)];
+            needed[3] = SpecialSymbols[random.Next(SpecialSymbols.Length)];
+
+            var fillerLength = random.Next(4, 9);
+            var filler = new String(Enumerable.Repeat(TotalAlphabet, fillerLength)
+                .Select(s => s[random.Next(s.Length)])
+                .ToArray());
+
+            var result = new char[fillerLength + 4];
```
Comment (djino97) [inline src/UserService.Business/GeneratePasswordCommand.cs:38]: Move this class in other file PasswordGenerationLogic

File: src/UserService.Business/GeneratePasswordCommand.cs [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+                {
+                    result[i] = filler[fillerIndex];
+                    fillerIndex++;
+                }
+                else
+                {
+                    result[i] = needed[neededIndex];
+                    neededIndex++;
+                }
+            }
```
Comment (djino97) [inline src/UserService.Business/GeneratePasswordCommand.cs:54]: move this logic in other private method

File: test/UserService.Business.UnitTests/GeneratePasswordCommandTests.cs (hunk 4 above)
Comment (djino97) [reply test/UserService.Business.UnitTests/GeneratePasswordCommandTests.cs:28]: no

**Review by djino97 — CHANGES_REQUESTED** [review]

File: src/UserService.Business/GeneratePasswordCommand.cs (hunk 7 above)
Comment (24eeeee) [reply src/UserService.Business/GeneratePasswordCommand.cs]: done

File: src/UserService.Business/GeneratePasswordCommand.cs (hunk 8 above)
Comment (24eeeee) [reply src/UserService.Business/GeneratePasswordCommand.cs]: done

**Review by djino97 — APPROVED** [review]

**Pull request merged**